app/src/main/java/com/parishod/watomatic/receivers/NotificationServiceRestartReceiver.kt

```diff
@@ -21,15 +22,19 @@ class NotificationServiceRestartReceiver : BroadcastReceiver() {
 
     private fun restartService(context: Context) {
         val preferencesManager = PreferencesManager.getPreferencesInstance(context)
-        if (preferencesManager.isForegroundServiceNotificationEnabled) {
+        if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.S || preferencesManager.isForegroundServiceNotificationEnabled) {
             val serviceIntent = Intent(context, KeepAliveService::class.java)
             // ToDo: Should probably start using foreground service to prevent IllegalState exception below
             try {
-                context.startService(serviceIntent)
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                    context.startForegroundService(serviceIntent)
```

Comment:
this will result in a crash for android 12+ .... you are not allowed to start a foreground service from the background

Reply:
actually it seems like it's possible after reboot ? https://developer.android.com/guide/components/foreground-services#background-start-restriction-exemptions